internal/txapp/txapp.go

```diff
@@ -306,9 +316,11 @@ func (r *TxApp) GetValidatorSetPower(ctx context.Context) (int64, error) {
 }
 
 // Execute executes a transaction.  It will route the transaction to the
-// appropriate module(s) and return the response.
+// appropriate module(s) and return the response. This method must only be
+// called from the consensus engine, sequentially, when executing transactions
+// in a block.
 func (r *TxApp) Execute(ctx TxContext, tx *transactions.Transaction) *TxResponse {
```

Comment:
We seem to have `TxContext` defined twice, once in `txapp` and once in `common`. Shouldn't we just ditch the one in `txapp` in favor of `common`?